Make PointSamplerBase much more robust in Parallel

friedmud opened the pull request:
My version of the fix @aeslaughter attempted here: #4468

Ensures that in the end only 1 processor is responsible for each point and that processor will be stable

Also splits the work up in parallel based on the geometric decomposition of the mesh so as you add more processors it will go faster.  Note that the geometric lookup is a loose bounding box.  It's not intended to be perfect...

This also works with ParallelMesh (I verified it) and handles all edge cases (that I can think of).

refs #4455

friedmud committed:
Make PointSamplerBase much more robust in Parallel - ensures that in the end only 1 processor is responsible for each point and that processor will be stable refs #4455

friedmud commented on framework/src/vectorpostprocessors/PointSamplerBase.C:
```diff
@@ -80,6 +98,31 @@ PointSamplerBase::execute()
+
+    unsigned int id = max_id[i];
+
+    if (id == pid)
```
Heh - yes it could.  I saw that too once I was looking at the diff on GitHub.  I think I had grander plans for `id` :smile: 

I'll change it

moosebuild commented:
Results of testing f57fd483babe7bdfdf9fc509bd35818b069e4e4d using moose_PR_test recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/10116

moosebuild commented:
Results of testing f57fd483babe7bdfdf9fc509bd35818b069e4e4d using moose_PR_test_dbg recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/10117

friedmud committed:
add a test for a point lying outside the domain refs #4455

moosebuild commented:
Results of testing 5bc9cc8f381d489ff1bc36de48a5ed3404e87c19 using moose_PR_pre_check recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/10181

friedmud commented:
Added a RunException test for a point lying outside the domain and fixed up a couple of small style issues...

This should be ready go to now if it passes...

moosebuild commented:
Results of testing 5bc9cc8f381d489ff1bc36de48a5ed3404e87c19 using moose_PR_test recipe:

Failed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/10182

moosebuild commented:
Results of testing 5bc9cc8f381d489ff1bc36de48a5ed3404e87c19 using moose_PR_test_dbg recipe:

Failed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/10183

moosebuild commented:
Results of testing 4bd8c17c052b3ff097ef9c72060ea6d750950cbe using moose_PR_pre_check recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/10184

moosebuild commented:
Results of testing 4bd8c17c052b3ff097ef9c72060ea6d750950cbe using moose_PR_test recipe:

Failed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/10185

friedmud commented:
Damn!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

permcody commented:
David says that people who use more than 3 exclamation points are insane. :smile:

friedmud commented:
That's pretty close to how I feel right now....

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

moosebuild commented:
Results of testing 4bd8c17c052b3ff097ef9c72060ea6d750950cbe using moose_PR_test_dbg recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/10186

moosebuild commented:
Results of testing 4bd8c17c052b3ff097ef9c72060ea6d750950cbe using moose_PR_test recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.com/view_job/10185
